src/models.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Common data structures used in this charm."""
```

Comment:
Comment/no action: I must confess that I was slightly confused with the file name "models" because it reminded me of the Juju Model concept.

Reply:
I suggested models as an option as it is quite pervasive in Django (and so in Python) and probably the most close to what those entities are.... types are also similar to the `typing` ... but sure, Juju models is also there. And rereading this sentence, entity does not sound very bad to me neither (but I like models more to be honest)

Reply:
Let's keep models for this PR. We can change it later